File: src/main/webapp/app/models/dashboard/boat-product-dashboard.ts
import { BoatLintReport } from '../lint-report';
import { BoatStatistics } from './boat-dashboard';
import { BoatCapability } from './boat-capability';

export interface BoatProductDashboard {
  portalKey: string;
  portalName: string;

  id: number;
  key: string;
  name: string;
  content: number;

  createdOn: Date;
  createdBy: String;

  capabilities: BoatCapability[];

  lastLintReport: BoatLintReport;

  statistics: BoatStatistics;
}

File: src/main/java/com/backbase/oss/boat/bay/web/views/dashboard/BoatProductDashboard.java
package com.backbase.oss.boat.bay.web.views.dashboard;

import com.backbase.oss.boat.bay.service.statistics.BoatStatistics;
import com.backbase.oss.boat.bay.web.views.lint.BoatLintReport;
import java.time.LocalDateTime;
import java.util.List;
import lombok.Data;

@Data
public class BoatProductDashboard {

    private String portalKey;
    private String portalName;

    private String id;
    private String key;
    private String name;
    private String content;

    private LocalDateTime createdOn;
    private String createdBy;

    private BoatLintReport lastLintReport;

    private BoatStatistics statistics;

}

